pkg/prove/proof.go

```diff
@@ -107,6 +117,49 @@ func txSharePosition(txs types.Txs, txIndex uint64) (startSharePos, endSharePos
 	return startSharePos, endSharePos, nil
 }
 
+// BlobShareRange returns the start and end positions for the shares
+// where a given message, referenced by its wrapped pfb transaction, was published at.
+// Note: only supports transactions containing a single message
+func BlobShareRange(tx types.Tx) (beginShare uint64, endShare uint64, err error) {
+	indexWrappedTx, isIndexWrapped := types.UnmarshalIndexWrapper(tx)
+	if !isIndexWrapped {
+		return beginShare, endShare, fmt.Errorf("not an index wrapped tx")
+	}
+
+	encCfg := encoding.MakeConfig(blobmodule.AppModuleBasic{})
+	decoder := encCfg.TxConfig.TxDecoder()
+
+	decodedTx, err := decoder(indexWrappedTx.Tx)
```

Comment:
[very optional]
feel free to ignore, but this is slightly more like the other times we use this function
```suggestion
	decodedTx, err := encCfg.TxConfig.TxDecoder()(indexWrappedTx.Tx)
```

Reply:
[b12d99e](https://github.com/celestiaorg/celestia-app/pull/1233/commits/b12d99e200e952b6e90e2ffc6e67fcf15456328e)